optuna/multi_objective/samplers/_motpe.py

```diff
@@ -640,7 +641,7 @@ def _get_observation_pairs(
 
         # Convert all objectives to minimization
         score = [
-            v if d == StudyDirection.MINIMIZE else -v  # type: ignore
+            cast(float, v) if d == StudyDirection.MINIMIZE else -v  # type: ignore
```

Comment:
[Just a question] Is this `# type: ignore` necessary? We may avoid it by using `cast(float, -v)`.

Reply:
Nice catch! I missed this as `# type: ignore` was placed here.
Mypy isn't happy with `cast(float, -v)`, because `Optional[float]` (`v`) doesn't support unary `-`.
51964b9  I changed here to use `-cast(float, v)` although it may feel strange.